tests/e2e/democracy_test.go

```diff
@@ -0,0 +1,257 @@
+	representativesTokens := map[string]sdk.Int{}
+
+	for _, representative := range stakingKeeper.GetAllValidators(s.consumerCtx()) {
+		currentRepresentativesRewards[representative.OperatorAddress] = sdk.NewDec(0)
+		nextRepresentativesRewards[representative.OperatorAddress] = sdk.NewDec(0)
+		representativesTokens[representative.OperatorAddress] = representative.GetTokens()
+	}
+
+	distrModuleAccount := distrKeeper.GetDistributionAccount(s.consumerCtx())
+	providerRedistributeAccount := authKeeper.GetModuleAccount(s.consumerCtx(), consumertypes.ConsumerToSendToProviderName)
+	//balance of consumer redistribute address will always be 0 when checked between 2 NextBlock() calls
+
+	currentDistrModuleAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), distrModuleAccount.GetAddress(), bondDenom).Amount)
+	currentProviderFeeAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), providerRedistributeAccount.GetAddress(), bondDenom).Amount)
+	currentCommunityPoolBalance := distrKeeper.GetFeePoolCommunityCoins(s.consumerCtx()).AmountOf(bondDenom)
+	for key := range currentRepresentativesRewards {
+		representativeAddr, _ := sdk.ValAddressFromBech32(key)
+		representativeReward := distrKeeper.GetValidatorOutstandingRewards(s.consumerCtx(), representativeAddr).Rewards.AmountOf(bondDenom)
+		currentRepresentativesRewards[key] = representativeReward
+	}
+
+	s.consumerChain.NextBlock()
+
+	nextDistrModuleAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), distrModuleAccount.GetAddress(), bondDenom).Amount)
+	nextProviderFeeAccountBalance := sdk.NewDecFromInt(bankKeeper.GetBalance(s.consumerCtx(), providerRedistributeAccount.GetAddress(), bondDenom).Amount)
+	nextCommunityPoolBalance := distrKeeper.GetFeePoolCommunityCoins(s.consumerCtx()).AmountOf(bondDenom)
+	for key := range nextRepresentativesRewards {
+		representativeAddr, _ := sdk.ValAddressFromBech32(key)
+		representativeReward := distrKeeper.GetValidatorOutstandingRewards(s.consumerCtx(), representativeAddr).Rewards.AmountOf(bondDenom)
+		nextRepresentativesRewards[key] = representativeReward
+	}
+
+	distrModuleDifference := nextDistrModuleAccountBalance.Sub(currentDistrModuleAccountBalance)
+	providerDifference := nextProviderFeeAccountBalance.Sub(currentProviderFeeAccountBalance)
+	communityPoolDifference := nextCommunityPoolBalance.Sub(currentCommunityPoolBalance)
+	representativeDifference := map[string]sdk.Dec{}
+	consumerRedistributeDifference := communityPoolDifference
+
+	for key, currentReward := range currentRepresentativesRewards {
+		representativeDifference[key] = nextRepresentativesRewards[key].Sub(currentReward)
+		consumerRedistributeDifference = consumerRedistributeDifference.Add(representativeDifference[key])
+	}
+
+	s.Require().Equal(distrModuleDifference, consumerRedistributeDifference)
+	s.Require().Equal(communityPoolDifference.Quo(consumerRedistributeDifference), distrKeeper.GetCommunityTax(s.consumerCtx()))
```

Comment:
Comment: "confirm that the percentage given to the community pool is equal to the configured community tax percentage."